src/Symfony/Component/Messenger/Bridge/Doctrine/Transport/Connection.php

```diff
@@ -432,6 +430,32 @@ protected function executeStatement(string $sql, array $parameters = [], array $
         return $stmt;
     }
 
+    private function executeInsert(string $sql, array $parameters = [], array $types = []): string
+    {
+        $this->driverConnection->beginTransaction();
+
+        try {
+            $this->driverConnection->executeStatement($sql, $parameters, $types);
+        } catch (TableNotFoundException $e) {
+            // create table
+            if ($this->autoSetup) {
+                $this->setup();
+            }
+
+            $this->driverConnection->executeStatement($sql, $parameters, $types);
+        }
+
+        $id = $this->driverConnection->lastInsertId();
+
+        if (false === $id) {
+            throw new TransportException('lastInsertId() returned false, no id was returned.');
+        }
+
+        $this->driverConnection->commit();
```

Comment:
shouldn't we rollback the transaction on exceptions too, instead of beginning a transaction and letting it uncommitted ?

Reply:
Another good question :) I was thinking we don't want to rollback anything, if the insert succeeds but lastInsertId() fails, then we want the insert to stay but let the lastInsertId() exception bubble up. I think moving the commit in to a finally might make sense?

Reply:
My concern is more about the state of the connection here: if an error happen (be it while executing the insert of getting the last insert id), we have started a transaction without ever finishing it.

Btw, an option would be to use the `transactional` method of DBAL, which takes care of ending the transaction with a rollback as appropriate (if we decide to rollback on any failure).

Reply:
Regarding keeping the insertion in case of failures when retrieving the last insert, I don't know what is the right behavior. I'm not sure the transport should persist the message while failing to report its id. I'll let someone knowing Messenger better than me decide on that point.

Reply:
@dunglas what are your thoughts on the behavior here?

My thought is that if the lastInsertId() call fails, I would still want to see the record in the database AND see the uncaught exception. But maybe a rollback is fine and me seeing the uncaught exception for the failure will be enough to investigate the problem.

As of right now I am always committing the transaction (in the finally block), even if something fails inside of the try.

Reply:
I changed the behavior so that the insert is rolled back on exception and the exception is rethrown.